File: Server/models/photoalbum.go
package models

import "go.mongodb.org/mongo-driver/bson/primitive"

type PhotoAlbum struct {
	ID            primitive.ObjectID   `json:"id" bson:"_id,omitempty"`
	CoverImageUrl string               `json:"cover_image_url" bson:"cover_image_url,omitempty"`
	Name          string               `json:"name" bson:"name,omitempty"`
	Description   string               `json:"description" bson:"description,omitempty"`
	Owner         primitive.ObjectID   `json:"owner" bson:"owner,omitempty"`
	VisibleTo     []primitive.ObjectID `json:"visible_to" bson:"visible_to,omitempty"`
}

File: Client/Android/app/src/main/java/com/photomanager/photomanager/main/home/api/model/Photoalbum.kt
package com.photomanager.photomanager.main.home.api.model

import com.photomanager.photomanager.main.home.api.ObjectId
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable

@Serializable
data class PhotoAlbum(
    @SerialName("id") val id: ObjectId,
    @SerialName("cover_image_url") val coverImageUrl: String,
    @SerialName("name") val name: String,
    @SerialName("description") val description: String,
    @SerialName("owner") val owner: ObjectId,
    @SerialName("visible_to") val visibleTo: List<ObjectId>
)